lib/ansible/modules/network/fortios/fortios_voip_profile.py

```diff
@@ -0,0 +1,1152 @@
+            rfc2543-branch: "disable"
+            rtp: "disable"
+            ssl-algorithm: "high"
+            ssl-auth-client: "<your_own_value> (source user.peer.name user.peergrp.name)"
+            ssl-auth-server: "<your_own_value> (source user.peer.name user.peergrp.name)"
+            ssl-client-certificate: "<your_own_value> (source vpn.certificate.local.name)"
+            ssl-client-renegotiation: "allow"
+            ssl-max-version: "ssl-3.0"
+            ssl-min-version: "ssl-3.0"
+            ssl-mode: "off"
+            ssl-pfs: "require"
+            ssl-send-empty-frags: "enable"
+            ssl-server-certificate: "<your_own_value> (source vpn.certificate.local.name)"
+            status: "disable"
+            strict-register: "disable"
+            subscribe-rate: "106"
+            unknown-header: "discard"
+            update-rate: "108"
+'''
+
+RETURN = '''
+build:
+  description: Build number of the fortigate image
+  returned: always
+  type: str
+  sample: '1547'
+http_method:
+  description: Last method used to provision the content into FortiGate
+  returned: always
+  type: str
+  sample: 'PUT'
+http_status:
+  description: Last result given by FortiGate on last operation applied
+  returned: always
+  type: str
+  sample: "200"
+mkey:
+  description: Master key (id) used in the last call to FortiGate
+  returned: success
+  type: str
+  sample: "id"
+name:
+  description: Name of the table used to fulfill the request
+  returned: always
+  type: str
+  sample: "urlfilter"
+path:
+  description: Path of the table used to fulfill the request
+  returned: always
+  type: str
+  sample: "webfilter"
+revision:
+  description: Internal revision number
+  returned: always
+  type: str
+  sample: "17.0.2.10658"
+serial:
+  description: Serial number of the unit
+  returned: always
+  type: str
+  sample: "FGVMEVYYQT3AB5352"
+status:
+  description: Indication of the operation's result
+  returned: always
+  type: str
+  sample: "success"
+vdom:
+  description: Virtual domain used
+  returned: always
+  type: str
+  sample: "root"
+version:
+  description: Version of the FortiGate
+  returned: always
+  type: str
+  sample: "v5.6.3"
+
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+
+fos = None
+
+
+def login(data):
+    host = data['host']
+    username = data['username']
+    password = data['password']
+
+    fos.debug('on')
+    if 'https' in data and not data['https']:
+        fos.https('off')
+    else:
+        fos.https('on')
+
+    fos.login(host, username, password)
+
+
+def filter_voip_profile_data(json):
+    option_list = ['comment', 'name', 'sccp',
+                   'sip']
+    dictionary = {}
+
+    for attribute in option_list:
+        if attribute in json and json[attribute] is not None:
+            dictionary[attribute] = json[attribute]
+
+    return dictionary
+
+
+def flatten_multilists_attributes(data):
+    multilist_attrs = []
+
+    for attr in multilist_attrs:
```

Comment:
I am not sure, that why are you trying to iterate over empty list, every time the fn shall get called.

Reply:
We did this intentionally to support different versions of FortiGate. Depending on the version there can be some attributes that change from one version to another. If these attributes end up being "multioptions" they need to be processed by the method "flatten_multilists_attributes" before they are sent to fortiosapi. The user can add the attributes to the list and this way we avoid generating another ansible module for each different version of fortigate.

Reply:
I have seen the discussion you had with my colleague @NilashishC, so If I understand it correctly you meant that user shall modify the `multilist_attrs` list from the code end directly and if that's true that's incorrect and should not be allowed, also I can see in the respective fn. takes `data` as fn. args but the same is being returned from the fn. without being operated. So, I would suggest to modify the fn. or remove it altogether.

Reply:
ok I will remove the code

Reply:
Thanks, once the PR shall be updated. I'll merge the code.

Reply:
Thanks @justjais , I have added a commit removing this code. Note I have also removed the use of global variables.